proposals/utils/stepper_helpers.py

```diff
@@ -68,25 +69,34 @@ def get_context_data(self, *args, **kwargs):
     def get_url(self):
         return ""
 
-    def get_errors(self):
+    def get_errors(self, include_children=False):
         return []
 
-    def css_classes(
-        self,
-    ):
-        classes = []
-        if self.is_current(self.stepper.request):
-            classes.append(
-                "active",
-            )
-        return " ".join(classes)
+    def get_ancestors(self):
+        """
+        Returns a list of all ancestors, including itself, from young to old.
+        """
+        ancestors = []
+        ancestors.append(self)
+        if not self.parent:
+            return ancestors
+        else:
+            parent = self.parent
+            ancestors.append(parent)
+            while parent.parent:
+                parent = parent.parent
+                ancestors.append(parent)
+        return ancestors
 
     def is_current(self, request):
         """
         Returns True if this item represents the page the user
-        is currently on.
+        is currently on and adds the active class to the item's
+        self.css_classes set. This gets called when building the stepper,
+        until is_current returns True.
         """
         if request.path == self.get_url():
+            self.css_classes.add("active")
```

Comment:
This method was previously stateless, being more of a "check" and not an action with consequences. If we add the CSS class here, we have to remember to do that for every time we write a custom `is_current()`, like in the Session item (which currently doesn't do this correctly).

I see how this makes sense, but I would prefer if the css class gets added in `is_current_check()` so we don't have so many different places to think about CSS classes.